wpilibj/src/main/java/org/wpilib/hardware/accelerometer/ADXL345_I2C.java

```diff
@@ -68,20 +67,62 @@ public enum Axes {
     }
   }
 
-  /** Container type for accelerations from all axes. */
-  @SuppressWarnings("MemberName")
-  public static class AllAxes {
-    /** Acceleration along the X axis in g-forces. */
-    public double XAxis;
-
-    /** Acceleration along the Y axis in g-forces. */
-    public double YAxis;
-
-    /** Acceleration along the Z axis in g-forces. */
-    public double ZAxis;
+  /**
+   * Container type for accelerations from all axes.
+   *
+   * @param x x-axis value
+   * @param y y-axis value
+   * @param z z-axis value
+   */
+  public record AllAxes(double x, double y, double z) implements StructSerializable {
+    /** Struct implementation for AllAxes. */
+    public static class AllAxesStruct implements Struct<AllAxes> {
```

Comment:
Should we move this into a separate file like the other struct definitions? I could see a case for this being a bit different, but it still feels odd to do this differently. (Ditto for C++)